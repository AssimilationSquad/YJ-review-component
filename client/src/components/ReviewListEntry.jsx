 const React = require('react');

class ReviewListEntry extends React.Component {
  constructor(props) {
    super(props);
  }
  render() {
    return (
      <div>
        <h3>This is one review</h3>
      </div>
    )
  }
}

module.exports = ReviewListEntry;