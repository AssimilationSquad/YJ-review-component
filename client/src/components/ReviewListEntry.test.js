import React from 'react';
import { renderer } from 'react-test-renderer';
import { mount } from 'enzyme';

import ReviewListEntry from './ReviewListEntry';

describe("LockScreen", () => {
  let props = undefined;
  let mountedReviewListEntry = undefined;
  const reviewListEntry = () => {
    if (!mountedReviewListEntry) {
      mountedReviewListEntry = mount(
        <ReviewListEntry {...props} />
      );
    }
    return mountedReviewListEntry;
  }

  beforeEach(() => {
    props = {
      revInfo: {
        Thumbnail: undefined,
        First_name: undefined,
        Review_date: undefined,
        Body:""
      },
      expand: undefined,
      handler: undefined
    };
    mountedReviewListEntry = undefined;
  });



  it ('always renders a div', () => {
    const divs = reviewListEntry().find("div");
    expect(divs.length).toBeGreaterThan(0);
  });

})

