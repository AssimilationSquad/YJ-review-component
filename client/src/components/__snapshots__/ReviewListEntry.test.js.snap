// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewListEntry renders the component correctly 1`] = `
<div
  className="reviewListEntry"
>
  <div
    className="img"
  >
    <img
      height="48"
      weight="48"
    />
  </div>
  <div
    className="name"
  />
  <div
    className="date"
  >
    November 2018
  </div>
  <div
    className="reviewBody"
  >
    
    <button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "visibility": "hidden",
        }
      }
    >
      ...Read more
    </button>
  </div>
</div>
`;
